File: sansa-query/sansa-query-spark/src/main/scala/net/sansa_stack/query/spark/rdd/op/NodeToSparkMapper.scala
package net.sansa_stack.query.spark.rdd.op

import org.aksw.jena_sparql_api.rdf.collections.NodeMapper
import org.apache.spark.sql.types.DataType

/** Interface for mapping Jena Node's to Java objects compatible with a certain spark datatype */
trait NodeToSparkMapper {
  def getSparkDatatype(): DataType
  def getNodeMapper(): NodeMapper[Object]
}

File: sansa-query/sansa-query-spark/src/main/java/net/sansa_stack/query/spark/rdd/op/NodeToSparkMapper.java
package net.sansa_stack.query.spark.rdd.op;

import org.aksw.jena_sparql_api.rdf.collections.NodeMapper;
import org.apache.spark.sql.types.DataType;

public interface NodeToSparkMapper {
    DataType getSparkDatatype();
    NodeMapper<?> getNodeMapper();
}
